xgboost_ray/main.py

```diff
@@ -918,7 +1000,7 @@ def train(params: Dict,
 
     start_actor_ranks = set(range(ray_params.num_actors))  # Start these
 
-    while tries <= max_actor_restarts:
+    while True:
```

Comment:
This could be problematic. If the actors run into some failure we cannot recover from, this will get into an infinite loop (I encountered a bug during testing in which I ran into this). 
`max_actor_restarts` is linked to non-elastic training, but I'd propose to either a) re-introduce this as a measure to control this behavior or b) introduce another parameter that limits the number of restarts in elastic training. Generally I'd prefer a) to keep the number of parameters low, but would like to hear your opinion on this.

Reply:
I think max_actor_restarts can be kept for this too.